titan/features/incar.py

```diff
@@ -0,0 +1,171 @@
+                stats["newReleaseHIV"] += 1
+
+        if self.active:
+            stats["incar"] += 1
+            if self.agent.hiv:
+                stats["incarHIV"] += 1
+
+    def incarcerate(self, model):
+        """
+        Incarcerate an agent or update their incarceration variables
+
+        args:
+            agent: agent being updated
+        """
+        hiv_bool = self.agent.hiv
+
+        if hiv_bool:
+            hiv_multiplier = self.agent.location.params.incar.hiv.multiplier
+        else:
+            hiv_multiplier = 1
+
+        # agent is incarcerated
+        if self.active:
+            self.time -= 1
+
+            # FREE AGENT
+            if self.time == 0:
+                self.add_new_release_agent(self.agent)
+                self.active = False
+
+                # become high risk on release
+                if (
+                    not self.agent.high_risk.active and model.params.features.high_risk
+                ):  # If behavioral treatment on and agent HIV, ignore HR period.
+                    self.agent.high_risk.become_high_risk()
+                    for bond in self.agent.location.params.high_risk.partnership_types:
+                        self.agent.mean_num_partners[
+                            bond
+                        ] += self.agent.location.params.high_risk.partner_scale
+                        self.agent.target_partners[bond] = utils.poisson(
+                            self.agent.mean_num_partners[bond], model.np_random
+                        )
+                    model.pop.update_partnerability(self.agent)
+
+                # does agent stay on haart
+                if hiv_bool:
+                    if self.agent.haart.active:
+                        if (
+                            model.run_random.random()
+                            <= self.agent.location.params.incar.haart.discontinue
+                        ):  # 12% remain surpressed
+                            self.agent.haart.active = False
+                            self.agent.haart.adherence = 0
+
+                        # END FORCE
+
+        # should the agent become incarcerated?
+        elif model.run_random.random() < (
+            self.agent.location.params.demographics[self.agent.race][
+                self.agent.sex_type
+            ].incar.prob
+            * hiv_multiplier
+            * model.calibration.incarceration
+        ):
+            incar_duration = self.agent.location.params.demographics[self.agent.race][
+                self.agent.sex_type
+            ].incar.duration.prob
+
+            bin = current_p_value = 1
+            p = model.run_random.random()
+            while p >= current_p_value:
+                current_p_value += incar_duration[bin].prob
+                bin += 1
+
+            timestay = model.run_random.randint(
+                incar_duration[bin].min, incar_duration[bin].max
+            )
+
+            self.active = True
+            self.time = timestay
```

Comment:
Is there a reason to define timestay and then actually use self.time? Intermediate variable seems unnecessary